test/parallel/test-path-dirname.js

```diff
@@ -57,3 +61,6 @@ assert.strictEqual(path.win32.dirname(''), '.');
 assert.strictEqual(path.win32.dirname('/'), '/');
 assert.strictEqual(path.win32.dirname('////'), '/');
 assert.strictEqual(path.win32.dirname('foo'), '.');
+
+assert.strictEqual(path.win32.dirname(url.pathToFileURL('c:\\')), 'c:\\');
+assert.strictEqual(path.win32.dirname(url.pathToFileURL('c:\\foo')), 'c:\\');
```

Comment:
```suggestion
if (common.isWindows) {
  assert.strictEqual(path.win32.dirname(url.pathToFileURL('c:\\')), 'c:\\');
  assert.strictEqual(path.win32.dirname(url.pathToFileURL('c:\\foo')), 'c:\\');
}
```
or
```suggestion
assert.strictEqual(path.win32.dirname('file:///c:/'), 'c:\\');
assert.strictEqual(path.win32.dirname('file:///c:/foo'), 'c:\\');
```
Unfortunately, `url.pathToFileURL()` is platform-specific here.